packer/table_transformations/csr_transformations.py

```diff
@@ -28,40 +33,52 @@ def from_obs_json_to_export_csr_df(obs_json: Dict) -> DataFrame:
     :return: data frame that has 4 (subject, diagnosis, biosource, biomaterial) index columns.
     The rest of columns represent concepts (aka variables)
     """
-
     df = ObservationSet(obs_json).dataframe
-    concept_pat_to_name = _concept_path_to_name(df)
+    df = transform_obs_df(df)
+    return df
 
-    # Transform sample and study data separately
+
+def transform_obs_df(df: DataFrame) -> DataFrame:
+    concept_pat_to_name = _concept_path_to_name(df)
+    # Transform sample and data outside of the sample hierarchy (study, radiology) separately
     sample_df = df
     study_df = None
+    radiology_df = None
+    if 'Radiology' in df.columns:
+        sample_df = df[df['Radiology'].isnull()]
+        sample_df.drop(columns=['Radiology'], inplace=True)
+        radiology_df = df[df['Radiology'].notnull()]
+        non_radiology_columns = [c for c in ID_COLUMN_MAPPING.keys() - [SUBJECT_ID_FIELD, 'Diagnosis', 'Radiology'] if
+                                 c in df.columns]
+        radiology_df.drop(columns=non_radiology_columns, inplace=True)
     if 'Study' in df.columns:
+        df = sample_df
         sample_df = df[df['Study'].isnull()]
         sample_df.drop(columns=['Study'], inplace=True)
         study_df = df[df['Study'].notnull()]
-        sample_columns = [c for c in ID_COLUMN_MAPPING.keys() - [SUBJECT_ID_FIELD, 'Study'] if c in df.columns]
-        study_df.drop(columns=sample_columns, inplace=True)
-    sample_df = from_obs_df_to_csr_df(sample_df)
-    df = sample_df
-    if study_df is not None:
-        study_df = from_obs_df_to_csr_df(study_df)
-        if sample_df.empty:
-            df = study_df
+        non_study_columns = [c for c in ID_COLUMN_MAPPING.keys() - [SUBJECT_ID_FIELD, 'Study'] if c in df.columns]
+        study_df.drop(columns=non_study_columns, inplace=True)
+    df = from_obs_df_to_csr_df(sample_df)
+    df = merge_non_hierarchical_entity_df(df, radiology_df, 'Radiology Id', ['Subject Id', 'Diagnosis Id'])
+    df = merge_non_hierarchical_entity_df(df, study_df, 'Study Id', ['Subject Id'])
+    df = df.rename(index=str, columns=concept_pat_to_name)
+    df = format_columns(df)
+    return df
+
+
+def merge_non_hierarchical_entity_df(df: DataFrame, entity_df: DataFrame, id_column: str, merge_columns: List[str]) -> DataFrame:
```

Comment:
`entity_df` can be `None`. You can make that explicit by using `Optional` from `typings`
```suggestion
def merge_non_hierarchical_entity_df(df: DataFrame, entity_df: Optional[DataFrame], id_column: str, merge_columns: List[str]) -> DataFrame:
```